### PR TITLE
Group labs and information are now visible on the main Results page

### PR DESCRIPTION
**Pull request by 0xf8f8ff**

### CODE REVIEW

**Comment by 0xf8f8ff** [comment]
Resolves #332

File: ci/run_tests_test.go [hunk 1]
```diff
@@ -50,7 +50,12 @@ func TestRunTests(t *testing.T) {
 		JobOwner: "muggles",
 	}
 
-	ed, err := runTests("scripts", &Docker{}, info, runData)
+	runner, err := NewDockerCI()
+	if err != nil {
+		t.Fatalf("failed to set up docker client: %v", err)
```
Comment (meling) [inline ci/run_tests_test.go:55]: Just err. Don’t need the extra details in the tests... the err will tell the same anyway

File: web/courses.go [hunk 2]
```diff
@@ -162,16 +162,16 @@ func (s *AutograderService) getAllLabs(request *pb.SubmissionsForCourseRequest)
 	case pb.SubmissionsForCourseRequest_GROUP:
 		enrolLinks = append(enrolLinks, s.makeGroupResults(course, assignments)...)
 	case pb.SubmissionsForCourseRequest_INDIVIDUAL:
-		enrolLinks = append(enrolLinks, makeResults(course, assignments)...)
+		enrolLinks = append(enrolLinks, makeResults(course, assignments, false)...)
 	default:
-		enrolLinks = append(makeResults(course, assignments), s.makeGroupResults(course, assignments)...)
+		enrolLinks = append(enrolLinks, makeResults(course, assignments, true)...) //append(makeResults(course, assignments), s.makeGroupResults(course, assignments)...)
```
Comment (meling) [inline web/courses.go:167]: Remove comment here?

Comment (0xf8f8ff) [reply web/courses.go]: Done.

File: ci/run_tests_test.go (hunk 1 above)
Comment (0xf8f8ff) [reply ci/run_tests_test.go]: Fixed.

File: public/src/components/teacher/Results.tsx [hunk 3]
```diff
@@ -143,6 +143,23 @@ export class Results extends React.Component<IResultsProps, IResultsState> {
             const ans = await this.props.onSubmissionStatusUpdate(selected);
             if (ans) {
                 selected.approvedDate = new Date().toLocaleString();
+
+                if (current && selected.groupid > 0) {
```
Comment (meling) [inline public/src/components/teacher/Results.tsx:147]: A few things in this func. As a random reviewer, it is a bit hard to follow. But it would help if `current` was named `currentSubmission` and `selected` was named `selectedSubmission`. And maybe adding a few comments (not many)... to explain what's going on.

File: public/src/components/teacher/Results.tsx [hunk 4]
```diff
@@ -143,6 +143,23 @@ export class Results extends React.Component<IResultsProps, IResultsState> {
...
+                    this.state.allSubmissions.forEach((e) => {
+                        if (e.enrollment.getGroup()?.getId() === selected.groupid) {
+                            e.labs.forEach((l) => {
+                                if (l.assignment.getId() === current.assignment.getId()) {
+                                    const double = l.submission;
```
Comment (meling) [inline public/src/components/teacher/Results.tsx:152]: What does `double` mean?

File: public/src/components/teacher/Results.tsx [hunk 5]
```diff
@@ -167,17 +184,23 @@ export class Results extends React.Component<IResultsProps, IResultsState> {
...
 
-    private getResultHeader(): string[] {
-        let headers: string[] = ["Name"];
-        headers = headers.concat(this.props.assignments.filter((e) => !e.getIsgrouplab()).map((e) => e.getName()));
+    private getResultHeader(): (string | JSX.Element)[] {
+        let headers: (string | JSX.Element)[] = ["Name", "Group"];
+        headers = headers.concat(this.props.assignments.map((e) => {
+            if (e.getIsgrouplab()) {
+                return <span style={{ whiteSpace: 'nowrap' }}>{e.getName() + " (g)"}</span>;
+            } else return e.getName()
```
Comment (meling) [inline public/src/components/teacher/Results.tsx:192]: Prefer `else { .. }`

File: web/courses.go [hunk 6]
```diff
@@ -184,14 +184,17 @@ func makeResults(course *pb.Course, assignments []*pb.Assignment) []*pb.Enrollme
...
+					subLink.Submission = sb
+				} else if addGroups && a.IsGroupLab && sb.GroupID > 0 && sb.GroupID == enrol.GroupID {
 					subLink.Submission = sb
 				}
 			}
 			allSubmissions = append(allSubmissions, subLink)
 		}
 
 		newLink.Submissions = allSubmissions
+
```
Comment (meling) [inline web/courses.go:197]: not need for new line here

File: public/src/components/teacher/Results.tsx (hunk 3 above)
Comment (0xf8f8ff) [reply public/src/components/teacher/Results.tsx:147]: Noted and updated.

File: public/src/components/teacher/Results.tsx (hunk 4 above)
Comment (0xf8f8ff) [reply public/src/components/teacher/Results.tsx:152]: Yeah, not the best name choice on my side. In general, as this is the main result page, all submissions are grouped by students. As a result, submissions for group assignments will have copies of that submission for each group member. Here we update all the copies of a submission with the changed state locally to avoid re-fetching the whole course from the server.
Changed name to `currentSubmissionCopy`, what do you think?

File: public/src/components/teacher/Results.tsx (hunk 5 above)
Comment (0xf8f8ff) [reply public/src/components/teacher/Results.tsx]: Done.

File: web/courses.go (hunk 6 above)
Comment (0xf8f8ff) [reply web/courses.go]: Fixed.

**Review by meling — APPROVED** [review]
